edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeAgentDockerModule.cs

```diff
@@ -19,11 +19,22 @@ public override bool Equals(IModule<DockerConfig> other)
         {
             if (ReferenceEquals(null, other))
                 return false;
+
             if (ReferenceEquals(this, other))
                 return true;
-            return string.Equals(this.Name, other.Name) &&
+
+            bool result = string.Equals(this.Name, other.Name) &&
```

Comment:
Should we modify like this as name, type and config are already covered in base.Equals method:

// Perform a full equality test in all cases except when a new Edge Agent container
// starts up and hasn't cached it's config yet
if (other is EdgeAgentDockerRuntimeModule otherRuntime && !otherRuntime.ConfigIsEmpty)
{
    return base.Equals(other);
}

return string.Equals(this.Name, other.Name) &&
    string.Equals(this.Type, other.Type) &&
    string.Equals(this.Config.Image, other.Config.Image);

Reply:
It's a choice between doing the downcast every time to ensure a single batch of string comparisons, or repeating the string comparisons occasionally (when agents with the same image--and represented by the same type--are being compared) and minimizing the downcasts. You'd really need to measure to decide which one is better, but I don't think this code is in a particularly hot path...